pkg/util/ansible_execution.go

```diff
@@ -64,14 +64,24 @@ func AnsibleExecution(
 	if err != nil && !k8serrors.IsNotFound(err) {
 		return err
 	}
+	nodeSetHash, err := nodeSet.GetSpecConfigHash()
+	if err != nil {
+		return err
+	}
+	annotations := map[string]string{
+		"nodesethash": nodeSetHash,
+	}
 	if ansibleEE == nil {
 		ansibleEE = &ansibleeev1.OpenStackAnsibleEE{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      executionName,
-				Namespace: deployment.GetNamespace(),
-				Labels:    labels,
+				Name:        executionName,
+				Namespace:   deployment.GetNamespace(),
+				Labels:      labels,
+				Annotations: annotations,
 			},
 		}
+	} else if ansibleEE.Annotations["nodesethash"] != annotations["nodesethash"] {
```

Comment:
The commit message needs to be updated to describe the new behavior. The implementation appears fine, but if you look at history it isn't clear it was added with this patch, or even why.

Which is another thing.